restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/ExtensionsConfigurationEndpoint.java

```diff
@@ -87,23 +88,44 @@ void init() {
      * @param responseType
      * @return
      */
-    protected Response getConfiguration(final String extensionId, final MediaType responseType) {
+    protected Response getConfiguration(final String extensionId, final Sid accountSid, final MediaType responseType) {
         //Parameter "extensionId" could be the extension Sid or extension name.
         if (!isSuperAdmin()) {
             throw new InsufficientPermission();
         }
 
         ExtensionConfiguration extensionConfiguration = null;
+        ExtensionConfiguration extensionAccountConfiguration = null;
+        Sid extensionSid = null;
+        String extensionName = null;
+
+        if(Sid.pattern.matcher(extensionId).matches()){
+            extensionSid = new Sid(extensionId);
+        } else {
+            extensionName = extensionId;
+        }
 
         if (Sid.pattern.matcher(extensionId).matches()) {
             try {
-                extensionConfiguration = extensionsConfigurationDao.getConfigurationBySid(new Sid(extensionId));
+                extensionConfiguration = extensionsConfigurationDao.getConfigurationBySid(extensionSid);
             } catch (Exception e) {
                 return status(NOT_FOUND).build();
             }
         } else {
             try {
-                extensionConfiguration = extensionsConfigurationDao.getConfigurationByName(extensionId);
+                extensionConfiguration = extensionsConfigurationDao.getConfigurationByName(extensionName);
+            } catch (Exception e) {
+                return status(NOT_FOUND).build();
+            }
+        }
+
+        if (accountSid!=null) {
```

Comment:
@abdulazizali77 I don't think this is proper here. As far as I understand, an extension can have a global configuration but also can have a configuration specific to an account. Thus we need to different methods, one to get the global configuration and one to get the configuration for a specific account